numbat/src/vm.rs

```diff
@@ -615,6 +649,60 @@ impl Vm {
                     };
                     self.push_quantity(result.map_err(RuntimeError::QuantityError)?);
                 }
+                op @ (Op::AddDateTime | Op::SubDateTime) => {
+                    let rhs = self.pop_quantity();
+                    let lhs = self.pop_datetime();
+
+                    // for time, the base unit is in seconds
+                    let base = rhs.to_base_unit_representation();
+                    let seconds_f = base.unsafe_value().to_f64();
+
+                    // TODO check that this handles negative durations correctly
+                    let duration = chrono::Duration::seconds(seconds_f.trunc() as i64)
+                        + chrono::Duration::nanoseconds(
+                            (seconds_f.fract() * 1_000_000_000f64).round() as i64,
+                        );
+
+                    self.push(Value::DateTime(
+                        match op {
+                            Op::AddDateTime => lhs + duration,
+                            Op::SubDateTime => lhs - duration,
+                            _ => unreachable!(),
+                        },
+                        chrono::Local::now().offset().fix(),
+                    ));
+                }
+                Op::DiffDateTime => {
+                    let unit = self.pop_quantity();
+                    let rhs = self.pop_datetime();
+                    let lhs = self.pop_datetime();
+
+                    let duration = lhs - rhs;
+                    // TODO error handling
+                    let duration = duration.num_nanoseconds().unwrap() as f64 / 1_000_000_000f64;
```

Comment:
`num_nanoseconds` returns its result as an `i64`. 2^63 nanoseconds are "only" 292 years, so currently we'll get a panic here if we attempt to subtract two datetimes that are further apart than that:
```
>>> now() - parse_datetime("1731-01-01T12:00:00Z")
thread 'main' panicked at numbat/src/vm.rs:685:63:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Ok good news, if we upgrade chrono, I think we can do this new function: 

```suggestion
                    let duration = lhs - rhs;
                    let duration = duration.subsec_nanos() as f64 / 1_000_000_000f64 + duration.num_seconds() as f64;
```